rust/lanzatool/src/generation.rs

```diff
@@ -94,6 +90,33 @@ impl fmt::Display for Generation {
     }
 }
 
+/// Read the kernel version from the name of a directory inside the toplevel directory.
+///
+/// The path looks like this: $toplevel/kernel-modules/lib/modules/6.1.1
+fn read_kernel_version(toplevel: &Path) -> Result<String> {
+    let path = fs::read_dir(toplevel.join("kernel-modules/lib/modules"))?
+        .into_iter()
+        .next()
+        .transpose()?
+        .map(|x| x.path())
+        .context("Failed to read path.")?;
+
+    let file_name = path
+        .file_name()
+        .and_then(|x| x.to_str())
+        .context("Failed to convert path to filename string.")?;
+
+    Ok(String::from(file_name))
+}
+
+fn read_build_time(toplevel: &Path) -> Result<String> {
+    Ok(
+        time::OffsetDateTime::from_unix_timestamp(fs::metadata(toplevel)?.mtime())?
```

Comment:
Is this even reliable?

Reply:
Why not? It's the same stuff the systemdboot builder does. We could do a default value (e.g ."Unknown").

Reply:
Made this default to "Unknown". If we realize this is an error we can change it later.